### PR TITLE
feat(Image): preview-src prop

### PR DESCRIPTION
**Pull request by wangcch**
Add `preview-src` to determine the original data.

```tsx
<Image
  src="https://zos.alipayobjects.com/rmsportal/jkjgkEfvpUPVyRjUImniVslZfWPnJuuZ.png?x-oss-process=image/auto-orient,1/resize,p_10/quality,q_10"
  preview={{
    src: 'https://zos.alipayobjects.com/rmsportal/jkjgkEfvpUPVyRjUImniVslZfWPnJuuZ.png',
  }}
/>;
```

Fixes https://github.com/react-component/image/pull/55
Fixes https://github.com/ant-design/ant-design/issues/28321

### CODE REVIEW

**Comment by shaodahong** [comment]
It does not resolve #54, #54 is like Carousel

File: src/Image.tsx [hunk 1]
```diff
@@ -52,6 +53,7 @@ const ImageInternal: CompoundedComponent<ImageProps> = ({
   height,
   style,
   preview = true,
+  previewSrc,
```
Comment (shaodahong) [inline src/Image.tsx:56]: This prop better put into `preview: { src }`

Comment (wangcch) [reply src/Image.tsx:56]: 👌ok

**Comment by wangcch** [comment]
> It does not resolve #54, #54 is like Carousel

https://github.com/react-component/image/pull/56/commits/837b4a2d643957e7db985106c29591535453e0b4 to close https://github.com/react-component/image/issues/54

File: src/Preview.tsx [hunk 2]
```diff
@@ -14,13 +14,16 @@ import { warning } from 'rc-util/lib/warning';
...
+import { ImageStatus } from './Image';
 
 const { useState, useEffect } = React;
 
 interface PreviewProps extends Omit<IDialogPropTypes, 'onClose'> {
   onClose?: (e: React.SyntheticEvent<Element>) => void;
   src?: string;
   alt?: string;
+  /** thumbnail src value */
+  thumbnail?: string;
```
Comment (shaodahong) [inline src/Preview.tsx:26]: If Preview use `thumbnail` too weird

https://user-images.githubusercontent.com/14831261/103391539-f4b87f00-4b54-11eb-9e95-493e8f19d573.mp4

Comment (wangcch) [reply src/Preview.tsx:26]: 😅 Maybe my thumbnails are a bit extreme.

Currently, the replacement is done when the image is loaded to achieve fast switching.
And thumbnail will appear on the first load and before the load is completed. After that it won't even appear.

Comment (shaodahong) [reply src/Preview.tsx:26]: Yep, but is not universal, load status maybe `Image` `<Spin />` or some text, like `placeholder` prop

Comment (wangcch) [reply src/Preview.tsx:26]: 🤔 expose the loading state? let me try

**Comment by shaodahong** [comment]
> > It does not resolve #54, #54 is like Carousel
> 
> [837b4a2](https://github.com/react-component/image/commit/837b4a2d643957e7db985106c29591535453e0b4) to close #54

No, #54 look like 


https://user-images.githubusercontent.com/14831261/103397093-b087a780-4b71-11eb-800d-eef71a097370.mp4

**Comment by wangcch** [comment]
> > > It does not resolve #54, #54 is like Carousel
> > 
> > 
> > [837b4a2](https://github.com/react-component/image/commit/837b4a2d643957e7db985106c29591535453e0b4) to close #54
> 
> No, #54 look like
> 
>  Kapture.2020-12-31.at.14.07.46.mp4

🤦‍♂️ I understand

File: src/Preview.tsx (hunk 2 above)
Comment (shaodahong) [reply src/Preview.tsx:26]: I mean this pr resolve Image src not same as preview src, ant other issues like Preview placeholder(if loading) we maybe create new or and Image placeholder is enough

Comment (wangcch) [reply src/Preview.tsx]: understand 👌
Not a good way. This PR will not process `thumbnails` / `load state`.
Hope to export `preview-src`.

eg: aliyun oss
```tsx
<Image
  src="https://xxx.xx/xxx.png?x-oss-process=image/auto-orient,1/resize,p_10/quality,q_10"
  preview={{
    src: 'https://xxx.xx/xxx.png',
  }}
/>;
```
Thanks 🙆‍♂️

**Review by shaodahong — APPROVED** [review]
LGTM

**Comment by shaodahong** [comment]
`+ rc-image@4.5.0`

Thanks

**Comment by shaodahong** [comment]
and help create pr in `antd` upgrade version(base branch `feature`), and related doc/example

**Pull request merged**